src/inference/src/ie_core.cpp

```diff
@@ -158,6 +158,14 @@ void stripDeviceName(std::string& device, const std::string& substr) {
 
 class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore> {
     mutable std::map<std::string, ov::InferencePlugin> plugins;
+    // to lock parallel access to pluginRegistry and plugins
+    mutable std::unordered_map<std::string, std::mutex> dev_mutexes;
+    std::mutex& get_mutex(const std::string& dev_name = "") const {
+        return dev_mutexes.at(dev_name);
```

Comment:
I'm just curious if exception thrown here it will be handled?

Reply:
We handle all exceptions. If we got an exception here it means that something wrong happened, because for all registered plugins we need to have mutex